File: web/src/types/scheduledTask.ts
export type ScheduledTaskState = 'idle' | 'running'
export type ScheduledTaskResult = 'running' | 'success' | 'failed' | 'skipped' | ''
export type ScheduledTaskTriggerSource = 'manual' | 'scheduled'

export interface ScheduledTask {
  id: string
  code: string
  name: string
  description: string
  cron_spec: string
  enabled: boolean
  registered: boolean
  run_timeout_seconds: number
  state: ScheduledTaskState
  next_run_at: string | null
  last_run_started_at: string | null
  last_run_finished_at: string | null
  last_result: ScheduledTaskResult | string
  last_error: string
  created_at: string
  updated_at: string
}

export interface ScheduledTaskRun {
  id: string
  task_code: string
  task_name: string
  trigger_source: ScheduledTaskTriggerSource
  status: ScheduledTaskResult
  summary: string
  result_payload: string
  error_message: string
  started_at: string
  finished_at: string | null
  duration_ms: number
  triggered_by_user_name: string
  triggered_by_user_id: string
  triggered_by_channel: string
  created_at: string
}

export interface ScheduledTaskRunPage {
  items: ScheduledTaskRun[]
  total: number
  page: number
  page_size: number
}

export interface UpdateScheduledTaskPayload {
  cron_spec?: string
  enabled?: boolean
  run_timeout_seconds?: number
}

File: internal/model/scheduled_task.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// ScheduledTask stores the persisted configuration and runtime snapshot of a registered task.
type ScheduledTask struct {
	ID                string     `gorm:"primaryKey;type:varchar(36)" json:"id"`
	Code              string     `gorm:"type:varchar(100);not null;uniqueIndex" json:"code"`
	Name              string     `gorm:"type:varchar(255);not null" json:"name"`
	Description       string     `gorm:"type:text;default:''" json:"description"`
	CronSpec          string     `gorm:"type:varchar(100);not null" json:"cron_spec"`
	Enabled           bool       `gorm:"not null;default:false" json:"enabled"`
	Registered        bool       `gorm:"not null;default:true" json:"registered"`
	RunTimeoutSeconds int        `gorm:"not null;default:300" json:"run_timeout_seconds"`
	NextRunAt         *time.Time `gorm:"index" json:"next_run_at"`
	LastRunStartedAt  *time.Time `json:"last_run_started_at"`
	LastRunFinishedAt *time.Time `json:"last_run_finished_at"`
	LastResult        string     `gorm:"type:varchar(20);default:''" json:"last_result"`
	LastError         string     `gorm:"type:text;default:''" json:"last_error"`
	CreatedAt         time.Time  `json:"created_at"`
	UpdatedAt         time.Time  `json:"updated_at"`
}

func (task *ScheduledTask) BeforeCreate(_ *gorm.DB) error {
	if task.ID == "" {
		id, err := newID()
		if err != nil {
			return err
		}
		task.ID = id
	}
	return nil
}
